command/operator_debug_test.go

```diff
@@ -275,136 +183,152 @@ func TestDebug_ClientToServer(t *testing.T) {
-			require.Contains(t, out, "Created debug archive")
-
-			// Reset buffers before next test
-			ui.OutputWriter.Reset()
-			ui.ErrorWriter.Reset()
-		})
+	var cases = testCases{
+		{
+			name:         "address=api, server-id=leader",
+			args:         []string{"-address", url, "-duration", "250ms", "-server-id", "leader"},
+			expectedCode: 0,
+			expectedOutputs: []string{
+				"Servers: (1/1)",
+				"Clients: (0/0)",
+				"Created debug archive",
+			},
+			expectedError: "",
+		},
+		{
+			name:         "address=api, server-id=all",
+			args:         []string{"-address", url, "-duration", "250ms", "-server-id", "all"},
+			expectedCode: 0,
+			expectedOutputs: []string{
+				"Servers: (1/1)",
+				"Clients: (0/0)",
+				"Created debug archive",
+			},
+			expectedError: "",
+		},
 	}
+
+	runTestCases(t, cases)
 }
 
-func TestDebugSuccesses(t *testing.T) {
+func TestDebug_Failures(t *testing.T) {
 	srv, _, url := testServer(t, false, nil)
 	defer srv.Shutdown()
 	testutil.WaitForLeader(t, srv.Agent.RPC)
 
-	ui := cli.NewMockUi()
-	cmd := &OperatorDebugCommand{Meta: Meta{Ui: ui}}
-
-	// NOTE -- duration must be shorter than default 2m to prevent testify from timing out
-
-	// Debug on the leader
-	code := cmd.Run([]string{"-address", url, "-duration", "250ms", "-server-id", "leader"})
-	assert.Equal(t, 0, code) // take note of failed return code, but continue to see why
-	assert.Empty(t, ui.ErrorWriter.String(), "errorwriter should be empty")
-	require.Contains(t, ui.OutputWriter.String(), "Starting debugger")
-	ui.OutputWriter.Reset()
-	ui.ErrorWriter.Reset()
+	var cases = testCases{
+		{
+			name:         "fails incorrect args",
+			args:         []string{"some", "bad", "args"},
+			expectedCode: 1,
+		},
+		{
+			name:         "Fails illegal node ids",
+			args:         []string{"-node-id", "foo:bar"},
+			expectedCode: 1,
+		},
+		{
+			name:         "Fails missing node ids",
+			args:         []string{"-node-id", "abc,def", "-duration", "250ms"},
+			expectedCode: 1,
+		},
+		{
+			name:         "Fails bad durations",
+			args:         []string{"-duration", "foo"},
+			expectedCode: 1,
+		},
+		{
+			name:         "Fails bad intervals",
+			args:         []string{"-interval", "bar"},
+			expectedCode: 1,
+		},
+		{
+			name:          "Fails bad address",
+			args:          []string{"-address", url + "bogus"},
+			expectedCode:  1,
+			expectedError: "invalid address",
+		},
+	}
 
-	// Debug on all servers
-	code = cmd.Run([]string{"-address", url, "-duration", "250ms", "-server-id", "all"})
-	assert.Equal(t, 0, code)
-	require.Empty(t, ui.ErrorWriter.String(), "errorwriter should be empty")
-	require.Contains(t, ui.OutputWriter.String(), "Starting debugger")
-	ui.OutputWriter.Reset()
-	ui.ErrorWriter.Reset()
+	runTestCases(t, cases)
 }
 
-func TestDebugFails(t *testing.T) {
+//////////////////////////////////////////////////////////////////////////////
+//////////////////////////////////////////////////////////////////////////////
```

Comment:
We don't typically have "banners" in our go code. I think this safe to leave out. Or if you want to keep it at least add some text under it like:

```go
// -------------------
// tests below don't use the test helper
```

Reply:
Removed